crates/ln-dlc-node/src/ldk_node_wallet.rs

```diff
@@ -180,58 +188,121 @@ where
         Ok(utxos)
     }
 
-    pub fn get_utxos_for_amount(
+    pub fn get_utxos_for_dlc_funding_transaction(
         &self,
         amount: u64,
-        lock_utxos: bool,
-        network: Network,
+        fee_rate: Option<u64>,
+        should_lock_utxos: bool,
     ) -> Result<Vec<Utxo>> {
-        let utxos = self.get_utxos()?;
-        // get temporarily reserved utxo from in-memory storage
+        let network = {
+            let bdk = self.bdk_lock();
+            bdk.network()
+        };
+
+        let fee_rate = fee_rate.map(|fee_rate| fee_rate as f32).unwrap_or_else(|| {
+            self.get_fee_rate(ConfirmationTarget::Normal)
```

Comment:
🤔 Our fee rate estimator is pretty pessimistic on the fees. `Normal` has been overpaid 2 times the other day. For the dlc channel offer we use a fee rate with a `ConfirmationTarget::Background`. Maybe we should use the same here?

Reply:
Let's switch to a better feerate estimator https://github.com/get10101/10101/pull/1916 .